### DISCUSSION TITLE
Advanced partitioning

### DISCUSSION THREAD

**Discussion by wernerzirkel** [yonileibowitz/yonileibowitz.github.io#5] [Ideas]
in the advanced partitioning, it is possible to use a certain column (e.g. _device_ id) + a timestamp to define the partition key. As shown in the [blog](https://yonileibowitz.github.io/blog-posts/data-partitioning.html), _device_ id is defined a string. From a database guy's perspective, it would be would be more natural to me to use integers here (can be internally _tostring(device_id)_ of course).

**Comment by y0nil** [comment]
Hi @wernerzirkel, and thank you for the feedback!

In Kusto, string indices are generally more detailed and efficient than numeric ones. 
Thus, if the purpose is to frequently filter on an ID column, and reach optimal performance, it'd be better to type that column as string.

Regardless, while there's no current support for partitioning on numeric columns, please feel free to submit this as a feature requests at https://aka.ms/adx.ideas.
